src/browser/preload_scripts.ts

```diff
@@ -76,38 +76,25 @@ export function fetchAndLoadPreloadScripts(
     const timer = new Timer();
     let result: Promise<LoadResponses>;
 
-    if (!preloadOption) {
-        preloadOption = [];
-    } else if (typeof preloadOption === 'string') {
-        // convert legacy `preloadOption` option into modern `preloadOption` option
-        preloadOption = [<PreloadInstance>{ url: preloadOption }];
-    }
-
-    if (!isPreloadOption(preloadOption)) {
-        const message = 'Expected `preload` option to be a string primitive OR an array of objects with `url` props.';
-        const err = new Error(message);
-        result = Promise.reject(err);
-    } else {
-        const loadedScripts: Promise<undefined>[] = preloadOption.map((preload: PreloadInstance) => {
-            // following if clause avoids re-fetch for remote resources already in memory
-            // todo: following if clause slated for removal (RUN-3227, blocked by RUN-3162), i.e., return always
-            if (
-                !REGEX_FILE_SCHEME.test(preload.url) && // not a local file AND...
-                System.getPreloadScript(preload.url)   // ...is already in memory?
-            ) {
-                // previously downloaded
-                logPreload('info', identity, 'previously cached:', preload.url);
-                updatePreloadState(identity, preload, 'load-succeeded');
-                return Promise.resolve(true);
-            } else {
-                // not previously downloaded *OR* previous downloaded failed
-                return fetchToCache(identity, preload).then(loadFromCache);
-            }
-        });
+    const loadedScripts: Promise<undefined>[] = preloadOption.map((preload: PreloadInstance) => {
+        // following if clause avoids re-fetch for remote resources already in memory
+        // todo: following if clause slated for removal (RUN-3227, blocked by RUN-3162), i.e., return always
+        if (
+            !REGEX_FILE_SCHEME.test(preload.url) && // not a local file AND...
```

Comment:
what if we just let it try to fetch an handle the fail?

Reply:
not to worry :) these lines are removed entirely by my next PR #181 (RUN-3321 electron.net).